Frontend/CarRentalAdmin/src/app/interceptors/authorization-interceptor.service.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { HttpRequest, HttpHandler, HttpEvent, HttpInterceptor } from '@angular/common/http';
+
+import { LocalStorageService } from '../services/local-storage.service';
+
+@Injectable()
+export class AuthorizationInterceptor implements HttpInterceptor {
+
+    constructor(private localStorageService: LocalStorageService) {
+    }
+
+    public intercept(
+        request: HttpRequest<any>,
+        next: HttpHandler
+    ): Observable<HttpEvent<any>> {
+        const modifiedRequest = this.Authorize(request);
+        return next.handle(modifiedRequest);
+    }
+
+    private Authorize(request: HttpRequest<any>): HttpRequest<any> {
+        const accessToken = this.localStorageService.getAccessToken();
+
+        if (!accessToken) {
+            return request;
+        }
+
+        return request.clone(
+            {
+                headers:
+                    request.headers.set(
+                        'Authorization',
+                        'Bearer ' + accessToken
+                    )
+            }
+        );
+    }
```

Comment:
```suggestion
        if (accessToken) {
            request.headers.set(
                'Authorization',
                'Bearer ' + accessToken
            )
        }

        return request;
    }
```

Can you do smth like this?
Why do you need to clone request?